### PR TITLE
Add FASTP complexity option

### LINKED ISSUE #78: Add fastp complexity filtering option
### Description of feature

Requested by @maxibor (and others), to reduce number of steps in pipeline for complexity filtering, with trade off of less powerful method.

### PR DESCRIPTION
**Pull request by jfy133**
Closes #78

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -54,7 +54,8 @@ process {
             params.shortread_clipmerge_skipadaptertrim ? "--disable_adapter_trimming" : "",
             params.shortread_clipmerge_adapter1 ? "--adapter_sequence ${params.shortread_clipmerge_adapter1}" : "",
             // filtering options
-            "--length_required ${params.shortread_clipmerge_minlength}"
+            "--length_required ${params.shortread_clipmerge_minlength}",
+            params.perform_shortread_complexityfilter && params.shortread_complexityfilter_tool == 'fastp' ? "--low_complexity_filter --complexity_threshold ${params.shortread_complexityfilter_fastp_threshold}" : ''
```
Comment (Midnighter) [inline conf/modules.config:58]: My natural tendency would be to put parentheses around this but it doesn't seem required.

```suggestion
            (params.perform_shortread_complexityfilter && params.shortread_complexityfilter_tool == 'fastp') ? "--low_complexity_filter --complexity_threshold ${params.shortread_complexityfilter_fastp_threshold}" : ''
```

File: docs/usage.md [hunk 2]
```diff
@@ -183,11 +183,11 @@ Complexity filtering can be activated via the `--perform_shortread_complexityfil
...
 
-There are currently two options for short-read complexity filtering: [`bbduk`](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/) and [`prinseq++`](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/).
+There are currently three options for short-read complexity filtering: [`bbduk`](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/), [`prinseq++`](https://github.com/Adrian-Cantu/PRINSEQ-plus-plus), and [`fastp`](https://github.com/OpenGene/fastp#low-complexity-filter).
 
 The tools offer different algorithms and parameters for removing low complexity reads. We therefore recommend reviewing the pipeline's [parameter documentation](https://nf-co.re/taxprofiler/parameters) and the documentation of both tools (see links above) to decide on optimal methods and parameters for your dataset.
```
Comment (Midnighter) [inline docs/usage.md:188]: More than two now

```suggestion
The tools offer different algorithms and parameters for removing low complexity reads. We therefore recommend reviewing the pipeline's [parameter documentation](https://nf-co.re/taxprofiler/parameters) and the documentation of the tools (see links above) to decide on optimal methods and parameters for your dataset.
```

**Review by Midnighter — APPROVED** [review]
Cool, thanks for that 🙂 Wasn't there something about taking a break or so to make time for other responsibilities? 😛

File: conf/modules.config (hunk 1 above)
Comment (jfy133) [reply conf/modules.config:58]: It is not required, but fine to add

**Comment by jfy133** [comment]
> Wasn't there something about taking a break or so to make time for other responsibilities? stuck_out_tongue

:shushing_face:

**Pull request merged**